File: examples/cpp/listener.cc
#include "talosos/logging.h"
#include "talosos/messages.h"
#include "talosos/node.h"

int main(int argc, char** argv) {
  talos::Init(argc, argv);

  auto node = talos::Node::Create("listener");

  auto sub = node->Subscribe<talos::msgs::String>(
      "/talker/chatter",
      [](const talos::msgs::String& msg) {
        TALOS_INFO("received: %s", msg.data.c_str());
      });

  TALOS_INFO("listener bound to %s", sub.key().c_str());
  node->Spin();
  return 0;
}

File: examples/python/listener.py
#!/usr/bin/env python3
"""Minimal Python subscriber — mirror of examples/cpp/listener.cc.

Pairs with talker.py (same process) or the C++ talker (cross-process).
"""

from __future__ import annotations

from talosos.messages import String
from talosos.runtime import Node, init


def main() -> int:
    init()
    node = Node.create("py_listener")

    def on_message(msg: String) -> None:
        print(f"received: {msg.data}")

    node.subscribe("/py_talker/chatter", String, on_message)
    print("py_listener listening on /py_talker/chatter")
    node.spin()
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
